Lib/test/test_os.py

```diff
@@ -3226,9 +3226,8 @@ def test_stat_inaccessible_file(self):
             self.skipTest("Unable to create inaccessible file")
 
         def cleanup():
-            # Give delete permission. We are the file owner, so we can do this
-            # even though we removed all permissions earlier.
-            subprocess.check_output([ICACLS, filename, "/grant", "Everyone:(D)"],
+            # Give delete permission to the owner (us)
```

Comment:
You're doing a lot more than you have to. Just `/deny` the right to synchronize on the file for users (`*BU`). `CreateFileW()` always requests `SYNCHRONIZE` access because it's in the API contract that File handles are waitable. OTOH, `DeleteFileW()` only needs `DELETE` access, so you can still delete the file without having to reset the security on it.

Reply:
It's only slightly more work, and this is a better emulation of the user report. It should be more robust if something else changes in the future (either failing hard, so we notice, or not failing at all). Being too targeted here could leave us open to the test passing being a false positive based on the environment, so I'd rather just go hard at it and be sure we're testing the right thing.